File: weeks/0/1/solution.cpp
#include <algorithm>
#include <iostream>
#include <string>

using namespace std;

constexpr int MAX_NUM_WORDS = int(1e6+1);

string words[MAX_NUM_WORDS];

void solve() {
  int numWords, wordPos; // N, K
  cin >> numWords >> wordPos;

  for (int i = 0; i < numWords; ++i) {
    cin >> words[i];
  }

  // 길이 오름차순으로 먼저 정렬하고, 그 다음 문자열을 사전순으로 정렬
  sort(begin(words), begin(words)+numWords, [](const string& w1, const string& w2) {
    return (w1.size() == w2.size()) ? (w1 < w2) : (w1.size() < w2.size());
  });

  string wordToFind = words[wordPos-1];
  cout << wordToFind << '\n';
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  solve();
}

File: weeks/0/1/solution.py
import sys
sys.setrecursionlimit(10**6)
readline = sys.stdin.readline

numWords, wordPos = map(int, readline().split())
words = [readline() for _ in range(numWords)]

# 길이 오름차순으로 먼저 정렬하고, 그 다음 문자열을 사전순으로 정렬
words.sort(key=lambda w: (len(w), w))

print(words[wordPos-1])
